code/game/objects/machinery/floodlight.dm

```diff
@@ -136,6 +136,131 @@
 /obj/machinery/floodlight/landing/attackby()
 	return
 
+/obj/machinery/floodlightcombat
+	name = "Armoured floodlight"
+	icon = 'icons/obj/machines/floodlight.dmi'
+	icon_state = "flood00"
+	anchored = TRUE
+	density = TRUE
+	/// Determines how much light does the floodlight make , every light tube adds 4 tiles distance.
+	var/Brightness = 0
+	var/On = 0
+	resistance_flags = UNACIDABLE
+	idle_power_usage = 50
+	active_power_usage = 2500
+	wrenchable = TRUE
+
+/obj/machinery/floodlightcombat/wrench_act(mob/living/user, obj/item/I)
+	. = ..()
+	to_chat(user , "You begin wrenching [src]'s bolts'")
+	if(!do_after(user, 2 SECONDS, TRUE, src))
+		return FALSE
+	if(anchored)
+		anchored = 0
+	else
+		anchored = 1
+	set_light(0)
+	On = 0
+
+
+/obj/machinery/floodlightcombat/proc/tip_over()
+	var/matrix/A = matrix()
+	density = FALSE
+	A.Turn(90)
+	transform = A
+	var/obj/item/light_bulb/tube/T = /obj/item/light_bulb/tube
+	for(T in src.contents)
+		T.status = 2
+	CalculateBrightness()
+
+/obj/machinery/floodlightcombat/Initialize()
+	. = ..()
+	start_processing()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/floodlightcombat/LateInitialize()
+	. = ..()
+	power_change()
+
+/obj/machinery/floodlightcombat/process()
+	if(On)
+		use_power(active_power_usage, EQUIP)
+	else
+		use_power(idle_power_usage, EQUIP)
+
+/obj/machinery/floodlightcombat/attack_alien(mob/living/carbon/xenomorph/M)
+	if(M.a_intent == INTENT_HARM)
+		var/obj/item/light_bulb/T = /obj/item/light_bulb/tube
+		var/list/lights = src.contents
+		if(lights.len == 0)
+			to_chat(M, "There are no lights to slash!")
+			return FALSE
+		to_chat(M, "You slash one of the lights!")
+		for(T in lights)
+			if(T.status > 0)
+				return FALSE
+			T.status = 2
+			CalculateBrightness()
+			break
+	if(M.a_intent == INTENT_DISARM)
```

Comment:
```suggestion
	else
```

Why make Help and Grab intents do nothing here and fail silently? Xenos on those intents may assume the lights are only damageable by acid.

Reply:
Code is preety barebones for now , i will just add the harm_intent interaction for help and grab. Vending checkes if the xeno is a crusher and scales that with the default tipping time,  which i might implement.